README.md

```diff
@@ -2,46 +2,101 @@
-[Click here to report an issue.](https://youtrack.jetbrains.com/newIssue?project=KT&clearDraft=true&c=Subsystems+Web+Site&c=subtask+of+KT-2555) If you need help with compiling or running the project locally, please join the #kontributors channel in the [Kotlin Slack](http://slack.kotlinlang.org), and we'll be happy to help you there.
+## Manual installation :whale:
+
+Start with command:
+
+```bash
+docker-compose up
+```
+
+The application will be available on `http://localhost:8080`.
+
+By default try.kotl.in is working on 8080 port.
+You can expose it to another port changing the config file 'docker-compose.yml'.
+```bash
+    ports:
+      - "your_port:8080"
+```
+
+So, if you need Google/Facebook/GitHub/Twitter of JetBrains account authorization
+you have to set key and secret_key in [configuration file](https://github.com/JetBrains/kotlin-web-demo/blob/master/docker/frontend/conf/Catalina/localhost/ROOT.xml).
 
-## Installation
-try.kotl.in is running on [Tomcat](https://tomcat.apache.org/). You can setup Tomcat locally using install_tomcat.xml.
-If you are installing tomcat manually, you need to download [ConnectorJ](http://dev.mysql.com/downloads/connector/j/) lib 
-and copy jar file to tomcat libraries.
+For example: 
+```xml
+    <Environment name="github_key" value="YOUR-KEY" type="java.lang.String" override="false"/>
+    <Environment name="github_secret" value="YOUR-SECRET-KEY" type="java.lang.String" override="false"/>
+```
 
-try.kotl.in frontend part uses bower as a dependency manager. To fetch this dependencies you have to install
- [node](https://nodejs.org/) and run `npm install` in kotlin.web.demo.frontend folder.
+## How to add your own courses :memo:
 
-You need MySql database to launch try.kotl.in, schema can be found [here](docker/db/files/init_db.sql).
+  - Add name of your course to [manifest.json](https://github.com/JetBrains/kotlin-web-demo/tree/master/kotlin.web.demo.server/examples).
+  - Add the folder with content near the [Examples folder](https://github.com/JetBrains/kotlin-web-demo/tree/master/kotlin.web.demo.server/examples)
+  with the same name as in manifest.json.
+  - Create folders with the topics of your course and don't forget about `manifest.json` with names of folders.
+  - Create 
+     1. Test.kt — file with test-methods
+     2. Task.kt — file with preview 
+     3. Solution.kt — file with solution
+     4. task.md - file with description of the task 
+     5. manifest.json - with 'junit' [configuration](https://github.com/JetBrains/kotlin-web-demo/blob/master/kotlin.web.demo.server/examples/Kotlin%20Koans/Introduction/Hello%2C%20world!/manifest.json)
+     
+   See example [Kotlin-Koans](https://github.com/JetBrains/kotlin-web-demo/tree/master/kotlin.web.demo.server/examples/Kotlin%20Koans) course.
+  
+## How to add your dependencies to kotlin compiler :books:
 
-## Building
-try.kotl.in is IDEA project. 
+How Kotlin-Web-Demo supports 3 compiler version:
```

Comment:
Don't think that we need enumeration of all supported versions here.

Reply:
@satamas You mean just a simple list?